src/helpers/EventBuilder.ts

```
@@ -130,4 +169,33 @@ export class EventBuilder {
   static getWeiForToken(address: string) {
     return Math.pow(10, tokenInfoCache.getDecimalsForToken(address))
   }
+
+  static async getNft(contractAddress: string, tokenId: string): Promise<Nft> {
+    try {
+      const url = `${GET_NFT_API_URL}?contractAddress=${contractAddress}&tokenId=${tokenId}`
+      const response = await fetch(url)
+      if (!response.ok)
+        throw new Error(`Failed to get NFT from ${url} - ${response.status}`)
+      const { result } = await response.json()
+      const { tokenUri, metadata, ownerAddress } = result
+      return {
+        tokenId,
+        contractAddress,
+        ownerAddress,
+        tokenUri,
+        metadata,
+      }
+    } catch (error) {
+      logger.error(
+        `Failed to get NFT for ${contractAddress} - ${tokenId} - ${error}`,
+      )
+      return {
```

it seems like this function will never throw an error, and instead will return a semi empty object when there is an error....i wonder if it's better to just let this function throw instead, since we are handling the error upstream anyway. i think it's probably sensible to let the consumer decide what to do with errors from this function, rather than always returning this manually created object which makes it harder for a consumer to know if the data is genuine or not

---

if this service is down, we should throw an error here. if we want to return a semi empty object, the caller of this function should handle the error and return the object. i think this function as it is, is unpredicatible since it swallows the error (depending on the use case, we may want to handle the error differently so it is more flexible to throw the error here)